components/matches_table/commons/matches_table.lua

```diff
@@ -0,0 +1,363 @@
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Countdown = require('Module:Countdown')
+local HiddenSort = require('Module:HiddenSort')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+local OpponentDisplay = OpponentLibraries.OpponentDisplay
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local UTC = ' <abbr data-tz="+0:00" title="Coordinated Universal Time (UTC)">UTC</abbr>'
+local WINNER_LEFT = 1
+local WINNER_RIGHT = 2
+local SCORE_STATUS = 'S'
+local DO_FLIP = true
+local NO_FLIP = false
+local LEFT_SIDE_OPPONENT = 'Left'
+local RIGHT_SIDE_OPPONENT = 'Right'
+local DEFAULT_QUERY_LIMIT = 1000
+
+-- If run in abbreviated roundnames mode
+local ABBREVIATIONS = {
+	["Upper Bracket"] = "UB",
+	["Lower Bracket"] = "LB",
+}
+
+---@class MatchesTable
+---@operator call(table<string, any>): MatchesTable
+---@field args table<string, any>
+---@field config table<string, boolean|string|number|string[]>
+---@field currentId string?
+---@field currentMatchHeader string[]?
+local MatchesTable = Class.new(function(self, args) self:init(args) end)
+
+---@param args any
+---@return MatchesTable
+function MatchesTable:init(args)
+	args = args or {}
+	self.args = args
+
+	args.tournament = args.tournament or mw.title.getCurrentTitle().prefixedText
+
+	self.config = {
+		limit = tonumber(args.limit) or DEFAULT_QUERY_LIMIT,
+		startDate = args.sdate,
+		endDate = args.edate,
+		section = args.section,
+		matchSection = args.matchsection,
+		showRound = not Logic.readBool(args.hideround),
+		sortRound = Logic.readBool(args.sortround),
+		showCountdown = Logic.readBool(args.countdown),
+		onlyShowExactDates = Logic.readBool(args.dateexact),
+		shortenRoundNames = Logic.readBool(args.shortedroundnames),
+		pages = Array.map(Array.extractValues(
+				Table.filterByKey(args, function(key) return key:find('^tournament%d-$') end)
+			), function(page) return (page:gsub(' ', '_')) end),
+	}
+
+	return self
+end
+
+---@return Html?
+function MatchesTable:create()
+	local matches = mw.ext.LiquipediaDB.lpdb('match2', {
+		limit = self.config.limit,
+		order = 'date asc',
+		conditions = self:buildConditions(),
+	})
+
+	if type(matches[1]) ~= 'table' then
+		return
+	end
+	self.matches = matches
+
+	self.output = mw.html.create('table')
+		:addClass('wikitable wikitable-striped sortable match-card')
+		:node(self:header())
+
+	Array.forEach(self.matches, function(match, matchIndex) return self:row(match) end)
```

Comment:
Why return inside the function? is this supposed to be an Array.map?
I personally don't really like that `self:row` expects the output to be setup already without having it as parameter.